src/vanilla/DanModel/DanCodeGenerator.cs

```diff
@@ -3933,25 +3940,299 @@ private static IEnumerable<CompositeType> CompositeTypeSubTypes(CompositeType co
+
         private static bool CompositeTypeIsExternalExtension(CompositeType compositeType)
-            => compositeType?.Extensions?.Get<bool>("x-ms-external") == true;
+            => GetBoolExtension(compositeType, "x-ms-external");
 
         internal static bool CompositeTypeIsAzureResourceExtension(CompositeType compositeType)
-            => compositeType?.Extensions?.Get<bool>(AzureExtensions.AzureResourceExtension) == true;
+            => GetBoolExtension(compositeType, AzureExtensions.AzureResourceExtension);
 
         internal static string SequenceTypeGetPageImplType(IModelType modelType)
-            => pageImplTypes.ContainsKey(modelType) ? pageImplTypes[modelType] : null;
+            => DictionaryGet(pageImplTypes, modelType);
 
         internal static void SequenceTypeSetPageImplType(IModelType modelType, string pageImplType)
+            => DictionarySet(pageImplTypes, modelType, pageImplType);
+
+        internal static MethodJva ResponseGetParent(Response response)
+            => DictionaryGet(responseParents, response);
+
+        internal static void ResponseSetParent(Response response, MethodJva parent)
+            => DictionarySet(responseParents, response, parent);
+
+        internal static bool ResponseIsPagedResponse(Response response)
+        {
+            MethodJva parent = ResponseGetParent(response);
+            return parent.IsPagingNextOperation || parent.IsPagingOperation;
+        }
+
+        internal static IModelType ResponseBodyClientType(Response response)
+            => ResponseBodyClientType(response, Settings.Instance);
+
+        private static IModelType ResponseBodyClientType(Response response, Settings settings)
+        {
+            IModelType result = GetIModelTypeResponseVariant(ResponseBodyWireType(response));
+            if ((IsFluent(settings) || IsAzure(settings)) && result is SequenceType bodySequenceType && ResponseIsPagedResponse(response))
+            {
+                SequenceType resultSequenceType = DependencyInjection.New<SequenceType>();
+                resultSequenceType.ElementType = bodySequenceType.ElementType;
+                SequenceTypeSetPageImplType(resultSequenceType, SequenceTypeGetPageImplType(bodySequenceType));
+                pagedListTypes.Add(resultSequenceType);
+                result = resultSequenceType;
+            }
+            return result;
+        }
+
+        private static void DictionarySet<TKey, TValue>(IDictionary<TKey, TValue> dictionary, TKey key, TValue value)
+        {
+            if (dictionary.ContainsKey(key))
+            {
+                dictionary[key] = value;
+            }
+            else
+            {
+                dictionary.Add(key, value);
+            }
+        }
+
+        private static TValue DictionaryGet<TKey, TValue>(IDictionary<TKey, TValue> dictionary, TKey key) where TValue : class
+            => dictionary.ContainsKey(key) ? dictionary[key] : null;
```

Comment:
This is useful, though. FWIW, `if (dict.TryGetValue("baz", out T val)) { ... }` works now, too.